### PR TITLE
solana support

### PR DESCRIPTION
**Pull request by a10zn8**

### CODE REVIEW

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/generic/GenericUpstream.kt [hunk 1]
```diff
@@ -191,7 +186,9 @@ open class GenericUpstream(
     }
 
     fun getIngressSubscription(): IngressSubscription {
-        return connector.getIngressSubscription()
+        val x = connector.getIngressSubscription()
+        println(x)
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/generic/GenericUpstream.kt:190]: Remove it?

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumChainSpecific.kt [hunk 2]
```diff
@@ -90,15 +90,6 @@ object EthereumChainSpecific : ChainSpecific {
         return EthereumLabelsDetector(reader, chain)
     }
 
-    override fun subscriptionTopics(upstream: GenericUpstream): List<String> {
-        val subs = if (upstream.getCapabilities().contains(Capability.WS_HEAD)) {
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumChainSpecific.kt:94]: We remove it, but now how can we get subscriptionTopics?

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/WsConnectionImpl.kt [hunk 3]
```diff
@@ -337,7 +337,7 @@ open class WsConnectionImpl(
         val status = subscriptionResponses.tryEmitNext(subscription)
         if (status.isFailure) {
             if (status == Sinks.EmitResult.FAIL_ZERO_SUBSCRIBER) {
-                log.debug("No subscribers to WS response")
+                log.debug("No subscribers to WS response - ${msg.value?.let { String(it) }}")
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/WsConnectionImpl.kt:340]: Could it be better to replace with placeholder {}?

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumChainSpecific.kt (hunk 2 above)
Comment (a10zn8) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumChainSpecific.kt:94]: It is here - io.emeraldpay.dshackle.upstream.ethereum.EthereumEgressSubscription#getAvailableTopics
I have found that this part war redundant and we can drop it. It was never used.

**Review by KirillPamPam — APPROVED** [review]

**Pull request merged**